File: Archive/AtCoder/AtCoder_Beginners_Selection/Traveling.nim
import strutils,sequtils
var
  N=stdin.readLine.parseInt
  lt,lx,ly,t,x,y:int
for i in 1..N:
  (t,x,y)=stdin.readLine.split.map(parseInt)
  if t mod 2!=(x+y) mod 2 or abs(x-lx)+abs(y-ly)>t-lt:
    echo "No"; quit()
  (lt,lx,ly)=(t,x,y)
echo "Yes"

File: Archive/AtCoder/AtCoder_Beginners_Selection/Traveling.py
n=int(input())
tx=[input() for _ in range(n)]
ext=exx=exy=0
for i in range(n):
    t,x,y=map(int,tx[i].split())
    if not(t-ext>=abs(x-exx)+abs(y-exy) and (t-ext-(abs(x-exx)+abs(y-exy)))%2==0):
        print("No")
        break
    ext,exx,exy=t,x,y
else:
    print("Yes")